litex/soc/integration/soc.py

```diff
@@ -41,6 +41,17 @@ def build_time(with_time=True):
     fmt = "%Y-%m-%d %H:%M:%S" if with_time else "%Y-%m-%d"
     return datetime.datetime.fromtimestamp(time.time()).strftime(fmt)
 
+def add_ip_constants(soc, name, ip):
+    _ip = ip.split(".")
+    assert len(_ip) == 4
+    for n in range(4):
+        assert int(_ip[n]) < 256
+        soc.add_constant(f"{name}{n+1}", int(_ip[n]))
+
+def add_mac_address_constants(soc, name, mac_address):
```

Comment:
Please add a minimal check as for IP address to ensure mac address is  < 2**48.

Reply:
done